src/WebServer_JSON.ino

```diff
@@ -3,7 +3,75 @@
 #include "src/Globals/Plugins.h"
 #include "StringProviderTypes.h"
 
+// ********************************************************************************
+// Web Interface get CSV value from task
+// ********************************************************************************
+void handle_csvval()
+{
+  String htmlData(25, '\0'); // Reserve for error message
+  const int printHeader = getFormItemInt(F("header"), 1);
+  bool printHeaderValid = true;
+  if (printHeader != 1 && printHeader != 0)
+  {
+    htmlData = F("ERROR: Header not valid!\n");
+    printHeaderValid = false;
+  }
+
+  const taskIndex_t taskNr    = getFormItemInt(F("tasknr"), INVALID_TASK_INDEX);
+  const bool taskValid = validTaskIndex(taskNr);
+  if (!taskValid)
+  {
+    htmlData = F("ERROR: TaskNr not valid!\n");
+  }
+
+  const int INVALID_VALUE_NUM = INVALID_TASKVAR_INDEX + 1;
+  const taskVarIndex_t valNr    = getFormItemInt(F("valnr"), INVALID_VALUE_NUM);
+  bool valueNumberValid = true;
+  if (valNr != INVALID_VALUE_NUM && !validTaskVarIndex(valNr))
+  {
+    htmlData = F("ERROR: ValueId not valid!\n");
+    valueNumberValid = false;
+  }
 
+  TXBuffer.startJsonStream();
+  if (taskValid && valueNumberValid && printHeaderValid)
+  {
+    const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(taskNr);
+
+    if (validDeviceIndex(DeviceIndex))
+    {
+      LoadTaskSettings(taskNr);
+      byte taskValCount = Device[DeviceIndex].ValueCount;
+      uint16_t stringReserveSize = (valNr == INVALID_VALUE_NUM ? 1 : taskValCount) * 24;
+      if (printHeader)
+      {
+        stringReserveSize += 11;
+      }
+      htmlData.reserve(stringReserveSize);
+
+      if (printHeader)
+      {
+        htmlData += "Name";
+        htmlData += ';';
+        htmlData += "Value";
+        htmlData += '\n';
```

Comment:
Every place where you add a `'\n'` is a good place to call `addHtml` and empty the string.
The general idea is to send the data to the connected client in chunks so we don't have to allocate a lot of memory.
It also makes the ESP very responsive as we generate the output at the same time the client downloads it.

Reply:
Ok, very good idea,  i will fix that :)